cypress/integration/vesselCommunications_spec.ts

```diff
@@ -0,0 +1,71 @@
+describe("As a beacon owner, I want to register my communication details so SAR can contact me in an emergency", () => {
+  const pageUrl = "/register-a-beacon/vessel-communications";
+
+  it("requires an MMSI number if the fixed VHF checkbox is selected", () => {
+    givenIAmOnTheVesselCommunicationsPage();
+    givenIHaveSelectedTheFixedVhfRadioOption();
+    andIHaveLeftTheRelevantTextInputBlank();
+
+    whenIClickContinue();
+
+    thenISeeAnError();
+  });
+
+  it("requires an MMSI number if the portable VHF checkbox is selected", () => {
+    givenIAmOnTheVesselCommunicationsPage();
+    givenIHaveSelectedThePortableVhfRadioOption();
+    andIHaveLeftTheRelevantTextInputBlank();
+
+    whenIClickContinue();
+
+    thenISeeAnError();
+  });
+
+  it("requires a phone number if the satellite telephone checkbox is selected", () => {
+    givenIAmOnTheVesselCommunicationsPage();
+    givenIHaveSelectedTheSatelliteTelephoneOption();
+    andIHaveLeftTheRelevantTextInputBlank();
+
+    whenIClickContinue();
+
+    thenISeeAnError();
+  });
+
+  it("requires a phone number if the satellite telephone checkbox is selected", () => {
+    givenIAmOnTheVesselCommunicationsPage();
+    givenIHaveSelectedTheMobileTelephoneOption();
+    andIHaveLeftTheRelevantTextInputBlank();
+
+    whenIClickContinue();
+
+    thenISeeAnError();
```

Comment:
It would be nice if we could assert that the user sees particular error messages in these tests. Currently, we're just checking the existence of the class on an element.

In the [NHS virtual visits](https://github.com/madetech/nhs-virtual-visit/blob/master/cypress/integration/admin/addingATrust.js#L81) they check for the existence of text within a HTML element which would be nice to do as it's being more specific about what error the user sees.

Reply:
I considered testing based on error messages, but discounted it on grounds of test-code coupling because copy changes would break the test suite.

At the moment we're just adding in error messages ad-hoc ("X field is required"), but when user research turns its attention to these they're likely to start changing quite quickly.  

I think the behaviour being tested here is whether an error message is shown, not that the text in error message is correct?

Reply:
I think I've been flipflopping with this too. Definitely makes sense to not couple the tests to things that could change pretty often.

But have also been worried that that could lead to not testing enough!

Reply:
Reviewed with @matthew-a-carr on a call.  Decided on a distinction when testing form errors:

1.  **Fields with one reason to result in an error**.  Simply test that an error message is displayed.
2. **Fields with many reasons to result in an error**.  Test that the _correct_ error message is displayed for the validation failure under test.

Reply:
Sorry @Ninamma looks like we were typing at the same time ^^!